src/coerce-functions.js

```diff
@@ -0,0 +1,51 @@
+import * as LogManager from 'aurelia-logging';
+
+export const coerceFunctions = {
+  none(a) {
+    return a;
+  },
+  number(a) {
+    const val = Number(a);
+    return !isNaN(val) && isFinite(val) ? val : 0;

```

Comment:
I would expect either NaN or Infinity to be returned if not a valid number, not 0.

Reply:
`NaN` doesn't work nicely with `===`, to handle it we need to sacrifice perf for more checks. With hundreds of bindings perf hit will be worse. I'm not sure what to do here. `Infinity`, on the other hand, still problematic but could be treated as is.

Reply:
I see it is a problem with two-way binding. Still, as a developer I would not expect 0 to be returned

Reply:
I think it could be application specific where `NaN` / `Infinity` are expected. For that case, we can always define another decorator via `createTypedObservable`:

```js
import {
  coerceFunctions,
  createTypedObservable
} from 'aurelia-binding'

coerceFunctions.$number = function(value) {
  return Number(value);
};
createTypedObservable('$number');
```

// usage
```js
class App {
  @observable.$number() numProp
}
```